geomstats/information_geometry/information_manifolds.py

```diff
@@ -0,0 +1,170 @@
+        point : array-like, shape=[..., dim]
+            Point representing a probability distribution.
+        x : array-like, shape=[...,]
+            Point where to evaluate the pdf
+        """
+        raise NotImplementedError("The pdf has not been hardcoded yet.")
+
+    def point_to_pdf(self, point):
+        """Compute pdf associated to point.
+
+        Compute the probability density function of the probability
+        distribution with parameters provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point representing a probability distribution.
+
+        Returns
+        -------
+        pdf : function
+            Probability density function of the probability distribution with
+            parameters provided by point.
+        """
+        geomstats.errors.check_belongs(point, self)
+        point = gs.to_ndarray(point, to_ndim=2)
+
+        def pdf(x):
+            """Generate parameterized function for pdf.
+
+            Parameters
+            ----------
+            x : array-like, shape=[n_points, dim]
+                Points at which to compute the probability
+                density function.
+
+            Returns
+            -------
+            pdf_at_x : array-like, shape=[..., n_points]
+                Values of pdf at x for each value of the parameters provided
+                by point.
+            """
+            raise NotImplementedError("The pdf method has not yet been implemented.")
+
+        return pdf
+
+    def point_to_cdf(self, point):
+        """Compute cdf associated to point.
+
+        Compute the cumulative density function of the probability
+        distribution with parameters provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point representing a probability distribution.
+
+        Returns
+        -------
+        cdf : function
+            Cumulative density function of the probability distribution with
+            parameters provided by point.
+        """
+        geomstats.errors.check_belongs(point, self)
+        point = gs.to_ndarray(point, to_ndim=2)
+
+        def cdf(x):
+            """Generate parameterized function for cdf.
+
+            Parameters
+            ----------
+            x : array-like, shape=[n_points, dim]
+                Points at which to compute the probability
+                density function.
+
+            Returns
+            -------
+            cdf_at_x : array-like, shape=[..., n_points]
+                Values of cdf at x for each value of the parameters provided
+                by point.
+            """
+            raise NotImplementedError("The cdf method has not yet been implemented.")
+
+        return cdf
+
+
+class FisherRaoMetric(RiemannianMetric):
+    """Class to derive the information metric from the pdf in InformationManifold."""
+
+    def __init__(self, pdf, support, dim):
```

Comment:
How about passing the information manifold as input to the constructor directly?

This would make the logic more consistent with invariant_metric.py i.e. a class of Riemannian metrics that are defined specifically on Lie groups. For which the lie group is passed in the constructor.
https://github.com/ninamiolane/geomstats/blob/master/geomstats/geometry/invariant_metric.py

Likewise, the Fisher rao metric is defined specifically for information manifolds. Thus the information manifold can be passed in the constructor.